Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,182 @@
+using Content.Server.Traitor.Uplink; //Refrences left in for future traitor PDA intergration
+using Content.Server.Traitor.Uplink.Components;
+using Content.Server.UserInterface;
+using Content.Shared.Interaction;
+using Content.Shared.PDA.Ringer;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Server.Player;
+using Robust.Shared.Player;
+using System;
+using System.Collections.Generic;
+
+namespace Content.Server.PDA.Ringer
+{
+    public enum Note
+        {
+            A,
+            Asharp,
+            B,
+            C,
+            Csharp,
+            D,
+            Dsharp,
+            E,
+            F,
+            Fsharp,
+            G,
+            Gsharp
+        }
+    public class RingerSystem : EntitySystem
+    {
+
+        private readonly Dictionary<string, string> _pathIndex = new Dictionary<string, string>();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            //General Event Subscriptions
+            SubscribeLocalEvent<RingerComponent, ComponentInit>(RandomizeRingtone);
+            //RingerBoundUserInterface Subscriptions
+            SubscribeLocalEvent<RingerComponent, RingerSetRingtoneMessage>(RingerUpdateRingtoneDriver);
+            SubscribeLocalEvent<RingerComponent, RingerPlayRingtoneMessage>(RingerPlayRingtone);
+            SubscribeLocalEvent<RingerComponent, RingerRequestUpdateInterfaceMessage>(UpdateRingerUserInterfaceDriver);
+
+            //Dictionary Path Addition
+            for(int indexCount = 0; indexCount < 12; indexCount++)
+            {
+                _pathIndex.Add(((Note)indexCount).ToString(), string.Concat("/Audio/Effects/RingtoneNotes/", ((Note)indexCount).ToString().Replace("sharp", "#").ToLower()) + ".ogg");
+            }
```

Comment:
Just use an array.
Also this is very ugly to read and could be cleaned up a lot.